### ISSUE TITLE
Missing generated/16px/paths in @blueprints/icons

### ISSUE THREAD

**Issue by dlech** [palantir/blueprint#5645]
#### Environment

- __Package version(s)__: @blueprints/icons v4.6.1
- __Operating System__: N/A
- __Browser name and version__: N/A

#### Actual behavior

CI for depedabot updates is failing due to missing files in the @blueprints/icons v4.6.1 package.

Example error message and stack trace from https://github.com/pybricks/pybricks-code/actions/runs/3191502668/jobs/5207890478#step:6:3181

```
    Cannot find module './generated/16px/paths' from 'node_modules/@blueprintjs/icons/lib/cjs/iconSvgPaths.js'

    Require stack:
      node_modules/@blueprintjs/icons/lib/cjs/iconSvgPaths.js
      node_modules/@blueprintjs/icons/lib/cjs/index.js
      node_modules/@blueprintjs/core/lib/cjs/components/icon/icon.js
      node_modules/@blueprintjs/core/lib/cjs/components/button/abstractButton.js
      node_modules/@blueprintjs/core/lib/cjs/components/button/buttons.js
      node_modules/@blueprintjs/core/lib/cjs/components/alert/alert.js
      node_modules/@blueprintjs/core/lib/cjs/components/index.js
      node_modules/@blueprintjs/core/lib/cjs/index.js
      test/index.tsx
      src/notifications/i18n.test.ts
```

Checking the `node_modules` folder on a local install, we can see that indeed there is no `generated/` folder. So it seems like this folder was not included in the build that was published to npm for some reason.

#### Expected behavior

New packages should not be broken :wink:

**Comment by sumeet07** [comment]
Similar issue while we build the application after getting above mentioned version of bp:
Failed to compile.
Module not found: Error: Can't resolve './generated/16px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconCodepoints.js 16:0-84
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconNames.js 20:0-72
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 17:0-57
Module not found: Error: Can't resolve './generated/16px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 18:0-57
Module not found: Error: Can't resolve './generated/20px/paths' in '/node_modules/@blueprintjs/icons/lib/esm'

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90
export 'IconSvgPaths16' (reexported as 'IconSvgPaths16') was not found in './iconSvgPaths' (possible exports: iconNameToPathsRecordKey)

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90
export 'IconSvgPaths20' (reexported as 'IconSvgPaths20') was not found in './iconSvgPaths' (possible exports: iconNameToPathsRecordKey)

**Comment by chrisbward** [comment]
Also getting this!

**Comment by bmoore15v** [comment]
Getting this error also! Just started today.

**Comment by Umenokin** [comment]
@dlech thank you!!! I created blueprintjs with vite the fist time struggling to find any working solution. Setting to prev versions made it working:
```
    "@blueprintjs/core": "4.10.2",
    "@blueprintjs/icons": "4.6.0",
```

**Comment by Violetdusk** [comment]
I was experiencing the same problem. I temporarily fixed this by specifying the `@blueprintjs/icons` version using selective dependency resolution.

Just add `resolutions` at the end of `package.json` like below. 

Note that this is just a temporary workaround.

```javascript
{
    // ...omitted
    "dependencies": { ... },
    "devDependencies": { ... }
    
    "resolutions": {
        "@blueprintjs/icons": "4.6.0"
    }
}

```

**Comment by adidahiya** [comment]
Thanks for raising the issue -- I'm looking into it now.

**Comment by mborik** [comment]
After your fixes, new versions released and clean install with them, still same error:
```
ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconCodepoints.js 16:0-84
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 17:0-72 17:0-72 17:0-72
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconNames.js 19:0-72
Module not found: Error: Can't resolve './generated/16px/blueprint-icons-16' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 18:0-40 18:0-40
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 17:0-57
Module not found: Error: Can't resolve './generated/16px/paths' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90 16:0-90 16:0-90 16:0-90
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

ERROR in ./node_modules/@blueprintjs/icons/lib/esm/iconSvgPaths.js 18:0-57
Module not found: Error: Can't resolve './generated/20px/paths' in '{workspacePath}/node_modules/@blueprintjs/icons/lib/esm'
 @ ./node_modules/@blueprintjs/icons/lib/esm/index.js 16:0-90 16:0-90 16:0-90 16:0-90
 @ ./node_modules/@blueprintjs/core/lib/esm/components/icon/icon.js 19:0-94 56:63-77 56:80-94 57:35-59
 @ ./node_modules/@blueprintjs/core/lib/esm/components/index.js 47:0-28 47:0-28
 @ ./node_modules/@blueprintjs/core/lib/esm/index.js 18:0-29 18:0-29
 @ ./src/index.tsx 11:0-52 17:41-56

webpack compiled with 4 errors
```
packages:
```
"@blueprintjs/core": "4.11.1"
"@blueprintjs/icons": "4.6.2"
"@blueprintjs/popover2": "1.7.1"
```

**Comment by adidahiya** [comment]
Acknowledged, I just noticed that too @mborik.

It doesn't seem like there's anything wrong with the build scripts (I see all the `/generated/` folders present when SSH'ing into the publish CI builds), but this problem may have been caused by the recent upgrade of Node from 16.14 -> 16.17. I will downgrade Node and try to publish again while I further debug the issue.

**Comment by adidahiya** [comment]
This is fixed in @blueprintjs/icons 4.6.3: https://unpkg.com/browse/@blueprintjs/icons@4.6.3/lib/cjs/generated/

I'll need to investigate why `node` 16.17 + `npm` 8.15 stopped publishing _just this one folder_ while `node` 16.14 + `npm` 8.5 works fine... 🤔

**Comment by adidahiya** [comment]
Unfortunately this is a problem again in @blueprintjs/icons 4.12.0 after the upgrade to npm v8.19.3 😢 

https://unpkg.com/browse/@blueprintjs/icons@4.12.0/lib/cjs/

**Comment by adidahiya** [comment]
This appears to be fixed in @blueprintjs/icons v4.12.1: https://unpkg.com/browse/@blueprintjs/icons@4.12.1/lib/cjs/

Still no idea why the `generated` folder was omitted from publishing with npm v8.15+ while a similar folder named `generated-icons` works fine... 🤷🏽
